pkg/monitors/uptimerobot/uptime-monitor.go

```diff
@@ -256,26 +256,109 @@ func (monitor *UpTimeMonitorService) processProviderConfig(m models.Monitor, cre
 
 			if providerConfig != nil && len(providerConfig.KeywordExists) != 0 {
-
 				if strings.Contains(strings.ToLower(providerConfig.KeywordExists), "yes") {
 					body += "&keyword_type=1"
 				} else if strings.Contains(strings.ToLower(providerConfig.KeywordExists), "no") {
 					body += "&keyword_type=2"
 				}
-
 			} else {
 				body += "&keyword_type=1" // By default 1 (check if keyword exists)
 			}
 
+			// Keyword Value (Required for keyword monitoring)
 			if providerConfig != nil && len(providerConfig.KeywordValue) != 0 {
-				body += "&keyword_value=" + providerConfig.KeywordValue
+				body += "&keyword_value=" + url.QueryEscape(providerConfig.KeywordValue)
 			} else {
 				log.Error(nil, "Monitor is of type Keyword but the `keyword-value` is missing")
 			}
 		}
 	} else {
 		body += "&type=1" // By default monitor is of type HTTP
 	}
+
+	// SubType (Optional for certain types)
+	if providerConfig != nil && len(providerConfig.SubType) != 0 {
+		body += "&sub_type=" + url.QueryEscape(providerConfig.SubType)
+	}
+
+	// Port (Optional for certain types)
+	if providerConfig != nil && providerConfig.Port > 0 {
+		body += "&port=" + strconv.Itoa(providerConfig.Port)
+	}
+
+	// Interval (Optional, in seconds)
+	if providerConfig != nil && providerConfig.Interval > 0 {
+		body += "&interval=" + strconv.Itoa(providerConfig.Interval)
+	} else {
+		body += "&interval=" + strconv.Itoa(DefaultInterval)
+	}
+
+	// Timeout (Optional, in seconds)
+	if providerConfig != nil && providerConfig.Timeout > 0 {
+		body += "&timeout=" + strconv.Itoa(providerConfig.Timeout)
+	}
+
+	// HTTP Auth (Optional)
+	if providerConfig != nil && len(providerConfig.HTTPAuthUsername) != 0 && len(providerConfig.HTTPAuthPassword) != 0 {
+		body += "&http_username=" + url.QueryEscape(providerConfig.HTTPAuthUsername)
+		body += "&http_password=" + url.QueryEscape(providerConfig.HTTPAuthPassword)
+		if providerConfig.HTTPAuthType > 0 {
+			body += "&http_auth_type=" + strconv.Itoa(providerConfig.HTTPAuthType)
+		}
+	}
+
+	// Post Type (Optional)
+	if providerConfig != nil && len(providerConfig.PostType) != 0 {
+		body += "&post_type=" + url.QueryEscape(providerConfig.PostType)
+	}
+
+	// Post Value (Optional)
+	if providerConfig != nil && len(providerConfig.PostValue) != 0 {
+		body += "&post_value=" + url.QueryEscape(providerConfig.PostValue)
+	}
+
+	// HTTP Method (Optional)
+	if providerConfig != nil && len(providerConfig.HTTPMethod) != 0 {
+		body += "&http_method=" + url.QueryEscape(providerConfig.HTTPMethod)
+	}
+
+	// Post Content Type (Optional)
+	if providerConfig != nil && len(providerConfig.PostContentType) != 0 {
+		body += "&post_content_type=" + url.QueryEscape(providerConfig.PostContentType)
+	}
+
+	// Alert Contacts (Optional)
+	if providerConfig != nil && len(providerConfig.AlertContacts) != 0 {
+		body += "&alert_contacts=" + url.QueryEscape(providerConfig.AlertContacts)
+	} else {
+		body += "&alert_contacts=" + url.QueryEscape(monitor.alertContacts)
```

Comment:
To be removed (duplicates of lines 231-235), generates an error: "Error: \"alert_contacts\" must be a string"